components/constellation/constellation.rs

```diff
@@ -183,6 +183,26 @@ struct Browser {
     session_history: JointSessionHistory,
 }
 
+/// A browsing context group.
+///
+/// https://html.spec.whatwg.org/multipage/#browsing-context-group
+#[derive(Clone, Default)]
+struct BrowsingContextGroup {
+    /// A browsing context group holds a set of top-level browsing contexts.
+    top_level_browsing_context_set: HashSet<TopLevelBrowsingContextId>,
+
+    /// The set of all event loops in this BrowsingContextGroup.
+    /// We store the event loops in a map
+    /// indexed by registered domain name (as a `Host`) to event loops.
+    /// It is important that scripts with the same eTLD+1,
+    /// who are part of the same browsing-context group
+    /// share an event loop, since they can use `document.domain`
+    /// to become same-origin, at which point they can share DOM objects.
+    event_loops: HashMap<Host, Weak<EventLoop>>,
+}
+
+type BrowsingContextGroupId = u32;
```

Comment:
Nit: can we make this a `struct BrowsingContextGroupId(u32)`?

Reply:
Ok, I also moved it to the `msg` crate...